drivers/scheduler/k8s/k8s.go

```diff
@@ -340,6 +342,90 @@ func (k *K8s) AddNewNode(newNode corev1.Node) error {
 	return nil
 }
 
+func (k *K8s) SetGkeConfig(kubeconfigPath string) error {
+
+	var clientConfig *rest.Config
+	var err error
+
+	if kubeconfigPath == "" {
+		clientConfig = nil
+	} else {
+		clientConfig, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	config, err := os.ReadFile(kubeconfigPath)
+	if err != nil {
+		return err
+	}
+	//configFile := base64.StdEncoding.EncodeToString(bdat)
+	// First parse the config
+	client, err := clientcmd.NewClientConfigFromBytes(config)
+	if err != nil {
+		return err
+	}
+
+	rawConfig, err := client.RawConfig()
+	if err != nil {
+		return err
+	}
+
+	// Then create a default client config with the default loading rules
+	client = clientcmd.NewDefaultClientConfig(rawConfig, &clientcmd.ConfigOverrides{})
+	if err != nil {
+		return err
+	}
+
+	cm, err := core.Instance().GetConfigMap("cloud-config", "default")
+	if err != nil {
+		log.Errorf("Error reading config map: %v", err)
+		return err
+	}
+
+	gkeJsonKey := cm.Data["cloud-json"]
+	cloudCred := backupapi.CloudCredentialObject{
+		Metadata: &backupapi.Metadata{
+			Name: "test",
+			Uid:  "newid",
+		},
+		CloudCredentialInfo: &backupapi.CloudCredentialInfo{
+			Type: backupapi.CloudCredentialInfo_Google,
+			Config: &backupapi.CloudCredentialInfo_GoogleConfig{
+				GoogleConfig: &backupapi.GoogleConfig{
+					ProjectId: "portworx-eng",
+					JsonKey:   gkeJsonKey,
+				},
+			},
+		},
+	}
```

Comment:
We should not do any backup operations here. This is a common function. Why do we need this?

Reply:
kubeauth.UpdateClientByCredObject(&cloudCred, clientConfig, &rawConfig) needs CloudCredentialObject which is avilable in px-backup-api

Reply:
We need to create our own function which will pick the values needed without needing the CloudCredentialInfo

Reply:
How will this be useful for non px-backup tests? Why will they create `CloudCredentialObject`

Reply:
Done